packages/tokamak/contracts-bedrock/src/L1/L1CrossDomainMessenger.sol

```diff
@@ -69,13 +69,27 @@ contract L1CrossDomainMessenger is CrossDomainMessenger, OnApprove, ISemver {
     /// @inheritdoc CrossDomainMessenger
     function _sendMessage(address _to, uint64 _gasLimit, uint256 _value, bytes memory _data) internal override {
         require(msg.value == 0, "Deny depositing ETH");
-        if (_value > 0) {
-            IERC20(nativeTokenAddress).safeTransferFrom(msg.sender, address(this), _value);
-            IERC20(nativeTokenAddress).approve(address(PORTAL), _value);
-        }
         PORTAL.depositTransaction(_to, _value, _gasLimit, false, _data);
     }
 
+    /// @notice unpack onApprove data
+    /// @param _data     Data used in OnApprove contract
+    function unpackOnApproveData(bytes calldata _data) public pure returns (address _to, uint32 _minGasLimit, bytes calldata _message) {
```

Comment:
Thank you always!
About unpackOnApproveData(onApprove), It seems only L1CDM can set the target Address. Is there any reason?

Reply:
Thank you!
Because for other function, it can be encoded into the message directly. The target in CDM is used for relayMessage function in L2.

Reply:
In optimism portal, I keep the same logic as previous version, only deposit native token.

Reply:
Oh, Thank you very much. I understood it thanks to your explanation.

@Zena-park, @nguyenzung 
How about removing onApprove in CDM? I think there is no receive function in legacy. I'm sorry about I'm missing about this.

Reply:
I think this function does not really matter if we add or not.

Reply:
@nguyenzung 
Thank you always!
I have a call with Zena.

As a result, I have a misunderstanding of onApprove. I'm sorry about that. First, I thought it is similar to receive function.

Our suggestion is below.

1. In unpackOnApproveData, unpack all argument of target function
- For example, in L1SB, we will use _initiateBridgeNativeToken as target function. So we can unpack the 
(address _from, address _to, uint256 _amount, uint32 _minGasLimit, bytes memory _extraData) is the all argument of function.
2. apply formatter (it seems to not apply formatter of solidity)

If you have another opinion, or there is any faults, please let me know ! 🙇🏼‍♂️

Thank you! 🙇🏼‍♂️

Reply:
@rlgns98kr Thank you Steven, let me update it. I also forgot formatting code.  

For encoded message, because owner and amount is already available, so should we only have _to, _minGasLimit and _extraData? I think it will reduce redundant checking and gas cost. In general, we only encode which params in the target function but not exist in onApprove()

Reply:
Thank you very much!

I think you are correct!
But in my thoughts, we may explain about approveAndCall to the frontend or users in the future. So if we unpack all arguments, we explain that more easily.

Is it okay? Is it a waste of gas..?

Reply:
It is okay, although will cost some more gas. Maybe it is easier for the frontend team.

Reply:
Thank you, Steven!
I updated